### ISSUE TITLE
Input as a DataFrame will have KeyError

### ISSUE THREAD

**Issue by john-mai-2605** [dreamquark-ai/tabnet#194]
**Describe the bug**

My data (X) are Panda dataframes, and labels (y) are NumPy arrays. Normally, sci-kit-learn estimators can handle this. However, TabNet seems to lack this feature (and accept only NumPy input).
Of course, I could just convert X to np array before feeding to TabNet, but it is much better if this is handled automatically.
**What is the current behavior?**
KeyError
**If the current behavior is a bug, please provide the steps to reproduce.**

**Expected behavior**

It should convert other data types to NumPy under the hood and return no error.
**Screenshots**

**Other relevant information:**
poetry version:  
python version: 
Operating System: 
Additional tools:

**Additional context**

**Comment by Optimox** [comment]
Hello @john-mai-2605,

It's definitely a feature that could be added, even though it does not seem super high priority to me, but I get that it can be annoying.

**Comment by jrfiedler** [comment]
In `abstract_model.py`, the change from `check_nan` to Scikit-learn's `check_array` ([this line](https://github.com/dreamquark-ai/tabnet/blob/develop/pytorch_tabnet/abstract_model.py#L147)) could make this fix easier.

The line
```python
        check_array(X_train)
```
could be changed to something like
```python
        X_train = check_array(X_train)
        y_train = check_array(y_train, ensure_2d=False, dtype=None)
        eval_set = [
            (
                check_array(subset[0]),
                check_array(subset[1], ensure_2d=False, dtype=None),
            )
            for subset in eval_set
        ]
```
which converts Pandas DataFrames and Series to Numpy arrays. There are a few other places that would need a similar change.

I tried the change above with the `census_example.ipynb` notebook, and it seems to work.

**Comment by Optimox** [comment]
@jrfiedler this would work for classification but not for multi regression where targets are 2D.

Moreover, if allowing pandas dataframes we would need to check that the columns are the same, with the same order etc.. I personally think that this feature is at the very edge of the scope of the library. It seems really easy to pass `df_train.values` when using the library. For that reason I don't see a strong value for this feature.
